data-products/src/main/scala/org/sunbird/analytics/util/TextBookUtils.scala

```diff
@@ -0,0 +1,183 @@
+package org.sunbird.analytics.util
+
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.ekstep.analytics.framework.util.{HTTPClient, JSONUtils}
+import org.sunbird.analytics.model.report.{ContentDetails, ContentInfo, dceTextbookReport, etbTextbookReport, FinalOutput, TenantInfo, TextBookDetails, TextBookInfo}
+import org.sunbird.cloud.storage.conf.AppConf
+
+case class etbTextbookData(channel: String, identifier: String, name: String, medium: String, gradeLevel: String,
+                             subject: String, status: String, createdOn: String, lastUpdatedOn: String, totalContentLinked: Integer,
+                             totalQRLinked: Integer, totalQRNotLinked: Integer, leafNodesCount: Integer, leafNodeUnlinked: Integer)
+
+case class dceTextbookData(channel: String, identifier: String, name: String, medium: String, gradeLevel:String, subject: String,
+                             createdOn: String, lastUpdatedOn: String, totalQRCodes: Integer, contentLinkedQR: Integer,
+                             withoutContentQR: Integer, withoutContentT1: Integer, withoutContentT2: Integer)
+
+object TextBookUtils {
+
+  def getTextBooks(config: Map[String, AnyRef], restUtil: HTTPClient): List[TextBookInfo] = {
+    val apiURL = Constants.COMPOSITE_SEARCH_URL
+    val request = JSONUtils.serialize(config.get("esConfig").get)
+    val response = restUtil.post[TextBookDetails](apiURL, request).result.content
+    response
+  }
+
+  def getTextbookHierarchy(textbookInfo: List[TextBookInfo],tenantInfo: RDD[TenantInfo],restUtil: HTTPClient)(implicit sc: SparkContext): (RDD[FinalOutput]) = {
+    var etbTextBookReport = List[etbTextbookData]()
+    var dceTextBookReport = List[dceTextbookData]()
+
+    textbookInfo.map(textbook => {
+      var apiUrl = AppConf.getConfig("hierarchy.search.api.url")+AppConf.getConfig("hierarchy.search.api.path")
+      if(textbook.status == "Live") { apiUrl = apiUrl+textbook.identifier }
+      else { apiUrl = apiUrl+textbook.identifier+"?mode=edit" }
+
+      val response = restUtil.get[ContentDetails](apiUrl)
+
+      if(response.params.status=="successful") {
+        val data = response.result.content
+
+        val etbTextBookResponse = generateETBTextbookReport(data)
+        etbTextBookReport = etbTextBookReport ++ etbTextBookResponse
+
+        val dceTextBookResponse = generateDCETextbookReport(data)
+        dceTextBookReport = dceTextBookReport ++ dceTextBookResponse
```

Comment:
Both of them are sequential, but can be parallized

Reply:
Done